examples/Transfer_Learning/basic_transfer_learning.py

```diff
@@ -0,0 +1,175 @@
+    "Training_Function": botorch_function_wrapper(Hartmann(dim=DIMENSION, negate=True)),
+}
+
+### Creating the Optimization Objective
+
+# Both test functions have a single output that is to be minimized.
+# The corresponding [Objective](baybe.objective.Objective)
+# is created as follows:
+
+objective = Objective(
+    mode="SINGLE", targets=[NumericalTarget(name="Target", mode="MIN")]
+)
+
+### Creating the Searchspace
+
+# The bounds of the search space are dictated by the test function:
+
+BOUNDS = Hartmann(dim=DIMENSION).bounds
+
+# First, we define one
+# [NumericalDiscreteParameter](baybe.parameters.numerical.NumericalDiscreteParameter)
+# per input dimension of the test function:
+
+discrete_params = [
+    NumericalDiscreteParameter(
+        name=f"x{d}",
+        values=np.linspace(lower, upper, POINTS_PER_DIM),
+    )
+    for d, (lower, upper) in enumerate(BOUNDS.T)
+]
+
+# ```{note}
+# While we could optimize the function using
+# [NumericalContinuousParameters](baybe.parameters.numerical.NumericalContinuousParameter),
+# we use discrete parameters here because it lets us interpret the percentages shown in
+# the final plot directly as the proportion of candidates for which there were target
+# values revealed by the training function.
+# ```
+
+# Next, we define a
+# [TaskParameter](baybe.parameters.categorical.TaskParameter) to encode the task context,
+# which allows the model to establish a relationship between the training data and
+# the data collected during the optimization process.
+# Because we want to obtain recommendations only for the test function, we explicitly
+# pass the `active_values` keyword.
+
+task_param = TaskParameter(
+    name="Function",
+    values=["Test_Function", "Training_Function"],
+    active_values=["Test_Function"],
+)
+
+# With the parameters at hand, we can now create our search space.
+
+parameters = [*discrete_params, task_param]
+searchspace = SearchSpace.from_product(parameters=parameters)
+
+# (Lookup)=
+### Generating Lookup Tables
+
+# We generate two lookup tables containing the target values of both test
+# functions at the given parameter grid.
+# Parts of one lookup serve as the training data for the model.
+# The other lookup is used as the loop-closing element, providing the target values of
+# the test functions on demand.
+
+grid = np.meshgrid(*[p.values for p in discrete_params])
+
+lookups: Dict[str, pd.DataFrame] = {}
+for function_name, function in test_functions.items():
+    lookup = pd.DataFrame({f"x{d}": grid_d.ravel() for d, grid_d in enumerate(grid)})
+    lookup["Target"] = lookup.apply(function, axis=1)
+    lookup["Function"] = function_name
+    lookups[function_name] = lookup
+lookup_training_task = lookups["Training_Function"]
+lookup_test_task = lookups["Test_Function"]
+
+### Simulation Loop
+
+# We now simulate campaigns for different amounts of training data unveiled,
+# to show the impact of transfer learning on the optimization performance:
```

Comment:
i wouldnt mind a bit more explanation here. Espcially the user here might be confused by the explanation just focusing on `different amounts of training data`. Why then are we drawing several samples for each fraction of training data in addition to that? -> to average out statistics via monte carlo technique

Reply:
Added something here, please resolve if gucci.